### ISSUE TITLE
Room booking: List previous requests / waitlist

### ISSUE THREAD

**Issue by olifre** [indico/indico#5786]
**Is your feature request related to a problem? Please describe.**
If room congestion happens and requests have to be denied, and later on, the accepted booking is cancelled by the booker, it would be nice to have a "waitlist" and allow other booking requests to take over the slot. 

**Describe the solution you'd like**
There are essentially two ways one could go about this:
* Implement an actual "waitlist", i.e. accept one pre-booking and put the others in a denied / waiting state. The room manager could even pre-acknowledge the waiting list such that if the approved booking is cancelled by the booker, the one from the waiting list is approved automatically. 
* Keep a history of booking requests for each room / timeslot, and automatically offer to the room manager to accept a previously denied booking in case an approved booking for the same room / time slot is cancelled by the booker. 

**Describe alternatives you've considered**
Essentially, this can be done manually by archiving the mails with pre-booking requests, keeping track of them, and then contacting users in case the accepted booking is cancelled / re-entering the denied booking for them. 
Of course, this is manual work and easily gets out of hand with many bookings. 

**Additional context**
This has actually been a feature idea by one of the room owners at our site, since it seems last-minute cancellation is a regular occurence ;-).

**Comment by ThiefMaster** [comment]
Hmm I first thought maybe https://github.com/indico/indico/projects/11#card-88312684 would cover the second part, but you want it by room, and not by user.

**Comment by olifre** [comment]
Indeed, from the room owner's point of view (i.e. if a meeting room is requested by multiple parties, one party wins, and then they cancel a few hours before the slot starts, so the other party could use it, after all), it would be ideal to have this by room. 
But I also see (different) use cases for the feature you linked ;-).
